### ISSUE TITLE
Error: could not find function \"new\"\n

### ISSUE THREAD

**Issue by MKLau** [End-to-end-provenance/RDataTracker#183]
I'm getting this error message. The script (*sensitivity.R*) fails to produce output.

```
library(devtools)
install_github('End-to-end-provenance/RDataTracker',ref='control2')
library(RDataTracker)
ddg.set.detail(1)
ddg.run('sensitivity.R',ddgdir='../data/ddg_sensitivity')
ddg.clear.detail()

```

**Comment by MKLau** [comment]
"environment" : {
"rdt:name" : "environment",
"rdt:architecture" : "x86_64",
"rdt:operatingSystem" : "unix",
"rdt:language" : "R",
"rdt:rVersion" : "R version 3.3.0 (2016-05-03)",
"rdt:script" : "/n/regal/ellison_lab/mklau/SarrSens/src/sensitivity.R",
"rdt:sourcedScripts" : ""
"rdt:scriptTimeStamp" : "2016-11-14T08.41.31EST",
"rdt:workingDirectory" : "/n/regal/ellison_lab/mklau/SarrSens/src",
"rdt:ddgDirectory" : "../data/ddg_sensitivity0",
"rdt:ddgTimeStamp" : "2016-11-14T09.03.59EST",
"rdt:rdatatrackerVersion" : "2.24.1",
"rdt:InstalledPackages": [
	{"package":"devtools", "version":"1.12.0"},
	{"package":"RDataTracker", "version":"2.24.1"}]
}},

**Comment by MKLau** [comment]
### Sensitivity
# parameters to manipulate

set.seed(12345)
source('../bin/PitcherPlantModel/src/global.R')
od <- '../results/sarsen/' # out directory and prefix

### Sensitivity

### define the parameter space for the simulation
days <- 30
fw <- round(sample(seq(0,75,length=100),1),1)
beta <- round(sample(seq(1e-06,2e-05,length=100),1),6)
d <- 5
k <- round(runif(1,0.0001,1),5)
m <- 0
n1 <- 3;n2 <- 5;n3 <- (days - (n1 + n2))
foodWeight <- c(rep(0,n1),rep(fw,n2),rep(0,n3))

ppSen <- pitcherPlantSim(days,foodWeight=foodWeight,beta=beta,d=d,k=k,m=m)

## a new comment

rng <- round(runif(1),10)*10^10
file <- paste(od,rng,sep='')
info <- c(rng,fw,beta,d,k,m)
info.f <- paste(od,'sarsen.info',sep='')
write.table(ppSen,file=file,row.names=FALSE,col.names=TRUE)
write.table(as.matrix(t(info)),file=info.f,row.names=FALSE,col.names=FALSE,append=TRUE)

**Comment by erboose** [comment]
Hi Matt,

This script ran without error on my laptop under Windows 7.  I modified the script to provide a complete path for "source" and "od", so perhaps the problem lies there.  Does the error occur if the provenance level is set to zero?

Emery

**Comment by MKLau** [comment]
Hi Emery, I get the same error message on my local machine (MacBook Air 2013 running MacOS 10.11.6). I can run the script just fine using source. I can also run ddg.run from the development branch, but if I use ddg.run from control2, I get that error message.

The \n in the error message suggest to me that there's an issue with string concatenation.

**Comment by erboose** [comment]
Hi Matt,

If I recall correctly there were some path issues that were corrected in development but not in control2. If you specify a complete path for "source" and "od", does the error still occur?

Emery

Sent from my iPhone

On Nov 16, 2016, at 1:02 PM, MKLau <notifications@github.com<mailto:notifications@github.com>> wrote:


Hi Emery, I get the same error message on my local machine (MacBook Air 2013 running MacOS 10.11.6). I can run the script just fine using source. I can also run ddg.run from the development branch, but if I use ddg.run from control2, I get that error message.

The \n in the error message suggest to me that there's an issue with string concatenation.

-

**Comment by MKLau** [comment]
Hi Emery, I'll check and see.

On Nov 16, 2016, at 1:17 PM, erboose <notifications@github.com<mailto:notifications@github.com>> wrote:

Hi Matt,

If I recall correctly there were some path issues that were corrected in development but not in control2. If you specify a complete path for "source" and "od", does the error still occur?

Emery

Sent from my iPhone

On Nov 16, 2016, at 1:02 PM, MKLau <notifications@github.com<mailto:notifications@github.com><mailto:notifications@github.com>> wrote:


Hi Emery, I get the same error message on my local machine (MacBook Air 2013 running MacOS 10.11.6). I can run the script just fine using source. I can also run ddg.run from the development branch, but if I use ddg.run from control2, I get that error message.

The \n in the error message suggest to me that there's an issue with string concatenation.

-
You are receiving this because you were assigned.
Reply to this email directly, view it on GitHub<https://urldefense.proofpoint.com/v2/url?u=https-3A__github.com_End-2Dto-2Dend-2Dprovenance_RDataTracker_issues_183-23issuecomment-2D261022541&d=CwMFaQ&c=WO-RGvefibhHBZq3fL85hQ&r=kbSzXk5uuINmNoo4MekzCE-8oe-IGwHYbug8SQuMN3U&m=yI3cLuiEoO2lHuC13GqGjJy1cHKiKgTOKPBkxGBfWlo&s=LXae9w6QCFKDFQAR5qk3mtcGnRSbsNt1zm33OI5rKgM&e=>, or mute the thread<https://urldefense.proofpoint.com/v2/url?u=https-3A__github.com_notifications_unsubscribe-2Dauth_AHHAK3cjlQtNFhSu6tbYzZETtqCePZLQks5q-2D0VDgaJpZM4KxW-2Do&d=CwMFaQ&c=WO-RGvefibhHBZq3fL85hQ&r=kbSzXk5uuINmNoo4MekzCE-8oe-IGwHYbug8SQuMN3U&m=yI3cLuiEoO2lHuC13GqGjJy1cHKiKgTOKPBkxGBfWlo&s=H4L9Eqmo1ZkaRg6jiSz2i3cAzyjGxRSGsL-Lb9Zd3p4&e=>.

-

**Comment by MKLau** [comment]
Hi Emery, even with specifying full paths from root, I still get this error. Let's get your branch merged with development and then try again.

**Comment by erboose** [comment]
Hi Matt,

Does the error occur with the current development version of RDT?  If not, I’ll do some more checking.

Thanks,
Emery

**Comment by MKLau** [comment]
Hi Emery, yeah, I set the level collection to 0.

Matt

On Nov 14, 2016, at 10:01 AM, erboose <notifications@github.com<mailto:notifications@github.com>> wrote:


Hi Matt,

This script ran without error on my laptop under Windows 7. I modified the script to provide a complete path for "source" and "od", so perhaps the problem lies there. Does the error occur if the provenance level is set to zero?

Emery
